### PR TITLE
Generate FromStr implementation for enums

### LINKED ISSUE #51: FromStr for core types
One of my use cases for this library would benefit from an `impl FromStr for EV_KEY` (to set a key from a command line argument, using the "clap" library). Implementations for this and various other types could be implemented by binding the relevant [libevdev helper functions](https://www.freedesktop.org/software/libevdev/doc/latest/group__misc.html). Alternatively, they could just be generated as part of a script like `make-event-names.py`.

I'd be able to put up a PR for this.

### PR DESCRIPTION
**Pull request by georgefst**
Closes #51.

### CODE REVIEW

**Comment by georgefst** [comment]
First four commits are just more tidying up of the script, unrelated to the purpose of this PR. I do wonder what that CI check is doing if it didn't catch any of this stuff...

File: src/enums.rs [hunk 1]
```diff
@@ -103,6 +125,29 @@ pub fn int_to_ev_rel(code: u32) -> Option<EV_REL> {
     }
 }
 
+impl std::str::FromStr for EV_REL {
+    type Err = ();
```
Comment (georgefst) [inline src/enums.rs:129]: I haven't written enough Rust to know what best practice is here. But there isn't really any useful error info we can hand back to the caller.

Comment (ndesh26) [reply src/enums.rs:129]: I agree there is not much that we can return to the user.

**Comment by ndesh26** [comment]
I should do something about the CI check. I need to stop treating the python scripts as secondary code and clean/tidy them regularly. Thanks for fixing all the issues with the script.

**Comment by georgefst** [comment]
FWIW, Pylint helped me catch a few things. By default it generates quite a lot of noise, but the integration with the VScode Python extension uses some sensible settings to silence a lot of the less useful stuff.

Then again, it's a small script that I can't imagine you'll ever need to change much, so it doesn't seem a huge issue for it not to be checked as rigorously as the Rust code.

**Pull request merged**